nncf/common/graph/traversal.py

```diff
@@ -0,0 +1,51 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from typing import TypeVar, Callable, List, Tuple, Optional, Any
+
+Node = TypeVar('Node')
+TraversingGraph = TypeVar('TraversingGraph')
+
+
+def traverse_graph(graph: TraversingGraph,
+                   traverse_function: Callable[[Node, List[Any]], Tuple[bool, List[Any]]],
+                   start_nodes: Optional[List[Node]] = None,
```

Comment:
Not clear in which order the traversal will start.

Reply:
Added the description